pyspedas/analysis/tests/tests.py

```diff
@@ -104,6 +110,109 @@ def test_subtract_average(self):
         subtract_average(['test1', 'test'], newname="testtest2")
         self.assertTrue(len(d[1]) == 6)
 
+    @unittest.skip('Currently subtract_average returns all NaN.')
+    def test_subtract_average_nan(self):
+        """Test subtract_average with NaN values."""
+        # Create a tplot variable with NaN values
+        store_data('test-nan', data={'x': [1., 2., 3., 4.], 'y': [3., 5., np.NaN, 8.]})
+
+        # Run subtract_average
+        tvar = subtract_average('test-nan')
+        d = get_data(tvar[0])
+
+        # Check that the result is not all NaN
+        self.assertFalse(np.isnan(d[1]).all(), "The result should not be all NaN.")
+
+        # Check that NaN remains where it was
+        self.assertTrue(np.isnan(d[1][2]), "NaN values should remain in the output.")
+
+        # Check that non-NaN values are properly adjusted
+        expected_values = np.array([3., 5., np.NaN, 8.]) - np.nanmean(np.array([3., 5., np.NaN, 8.]))
+        np.testing.assert_array_almost_equal(d[1], expected_values,
+                                             err_msg="Non-NaN values should be adjusted by subtracting the mean.")
+
+        # Additional NaN scenarios
+        store_data('test-nan-all', data={'x': [1., 2., 3., 4.], 'y': [np.NaN, np.NaN, np.NaN, np.NaN]})
+
+        # Check all NaN case
+        result_all = subtract_average('test-nan-all')
+        d_all = get_data(result_all[0])
+        self.assertTrue(np.isnan(d_all[1]).all(), "All NaN input should result in all NaN output.")
+
+    def test01_subtract_average_with_suffix(self):
```

Comment:
With some small changes, many tests in this group could be parameterized to take a user-specified function, so they could be reused for other tplot routines that operate on tplot variables one at a time.

Reply:
True. I think when tplot will be merged with pyspedas, we can open an issue to improve testing of pytplot routines.

Reply:
The refactoring to make the tests reusable doesn't need to happen right away, so this is good to merge as-is.